pytorch_lightning/trainer/properties.py

```diff
@@ -185,7 +185,21 @@ def progress_bar_dict(self) -> dict:
         """ Read-only for progress bar metrics. """
         ref_model = self.model if not self.data_parallel else self.model.module
         ref_model = cast(LightningModule, ref_model)
-        return dict(**ref_model.get_progress_bar_dict(), **self.logger_connector.progress_bar_metrics)
+
+        standard_metrics = ref_model.get_progress_bar_dict()
+        logged_metrics = self.logger_connector.progress_bar_metrics
+        duplicates = list(standard_metrics.keys() & logged_metrics.keys())
+        if duplicates:
+            rank_zero_warn(
+                f"The progress bar already tracks a metric with the name '{duplicates[0]}' and"
+                f" `self.log('{duplicates[0]}', ..., prog_bar=True)` will overwrite this value. "
```

Comment:
shall it have all the duplicate keys? for eg `v_num` and `split_idx` are also there

Reply:
it has nothing to do with v_num and split_idx. 
Here I'm just simplifying the message to one occurence. For 2 duplicates, the user would see the first one and then after fixing it the second time the next message

Reply:
not even doing it this way `f"with names {', '.join(duplicates)} and self.log('duplicates[0]', ...)"`, just to avoid the possibility of second time? one can override `get_progress_bar_dict` hook and add a key there and also log it somewhere else too. Although this case is super rare.

Reply:
ok I did it